app/src/main/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/RecordingController.kt

```diff
@@ -0,0 +1,58 @@
+package co.netguru.baby.monitor.client.feature.voiceAnalysis
+
+import co.netguru.baby.monitor.client.common.RunsInBackground
+import co.netguru.baby.monitor.client.feature.machinelearning.MachineLearning
+import co.netguru.baby.monitor.client.feature.noisedetection.NoiseDetector
+import io.reactivex.Observable
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+import javax.inject.Inject
+
+class RecordingController @Inject constructor(
+    private val aacRecorder: AacRecorder
+) {
+    var voiceAnalysisOption: VoiceAnalysisOption = VoiceAnalysisOption.MACHINE_LEARNING
+    private var rawData = emptyArray<Byte>()
+    private var newData = emptyArray<Short>()
+
+    fun startRecording(): Observable<RecordingData> {
+        return aacRecorder.startRecording()
+            .doOnNext {
+                addData(it)
+            }
+            .map {
+                val data = when {
+                    voiceAnalysisOption == VoiceAnalysisOption.MACHINE_LEARNING &&
+                            newData.size >= MachineLearning.DATA_SIZE -> RecordingData.MachineLearning(
+                        rawData.toByteArray(),
+                        newData.toShortArray()
+                    )
+                    voiceAnalysisOption == VoiceAnalysisOption.NOISE_DETECTION &&
+                            newData.size >= NoiseDetector.DATA_SIZE -> RecordingData.NoiseDetection(
+                        newData.takeLast(NoiseDetector.DATA_SIZE).toShortArray()
+                    )
+                    else -> RecordingData.Raw(it)
+                }
+                if (newData.size >= MachineLearning.DATA_SIZE) clearRecordingData()
+                data
+            }
+    }
+
+    @RunsInBackground
+    fun addData(array: ByteArray) {
+        newData = newData.plus(bytesToShorts(array).toTypedArray())
+        rawData = rawData.plus(array.toTypedArray())
+    }
+
+    @RunsInBackground
+    private fun bytesToShorts(bytes: ByteArray): ShortArray {
+        val shorts = ShortArray(bytes.size / 2)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/92c7fea843a9bd05df3c694ba8c05996f0ca5e22/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/RecordingController.kt#L49"></span>This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>